proton

```diff
@@ -127,6 +138,85 @@ def try_get_game_library_dir():
 
     return None
 
+# Function to find the installed location of DLL files for use by WINE/Proton
+# from the NVIDIA Linux driver
+#
+# See https://gitlab.steamos.cloud/steamrt/steam-runtime-tools/-/issues/71 for
+# background on the chosen method of DLL discovery.
+#
+# On success, returns a str() of the absolute-path to the directory at which DLL
+# files are stored
+#
+# On failure, returns None
+def find_nvidia_wine_dll_dir():
+    # Load libdl and libGLX_nvidia off the bat, if this fails don't waste time on
+    # the later stuff
+    libdl = CDLL(find_library("libdl"))
+    libglx_nvidia = CDLL("libGLX_nvidia.so.0")
+
+    if libdl is None or libglx_nvidia is None:
+        return None
+
+    # from dlinfo(3)
+    #
+    # struct link_map {
+    #     ElfW(Addr) l_addr;  /* Difference between the
+    #                            address in the ELF file and
+    #                            the address in memory */
+    #     char      *l_name;  /* Absolute pathname where
+    #                            object was found */
+    #     ElfW(Dyn) *l_ld;    /* Dynamic section of the
+    #                            shared object */
+    #     struct link_map *l_next, *l_prev;
+    #                         /* Chain of loaded objects */
+    #
+    #     /* Plus additional fields private to the
+    #        implementation */
+    # };
+    RTLD_DI_LINKMAP = 2
+    class link_map(Structure):
+        _fields_ = [("l_addr", c_void_p), ("l_name", c_char_p), ("l_ld", c_void_p)]
+
+    # from dlinfo(3)
+    #
+    # int dlinfo (void *restrict handle, int request, void *restrict info)
+    dlinfo_func = libdl.dlinfo
+    dlinfo_func.argtypes = c_void_p, c_int, c_void_p
+    dlinfo_func.restype = c_int
+
+    # Allocate a link_map object
+    glx_nvidia_info_ptr = POINTER(link_map)()
+
+    # Run dlinfo(3) on the handle to libGLX_nvidia.so.0, storing results at the
+    # address represented by glx_nvidia_info_ptr
+    if dlinfo_func(libglx_nvidia._handle,
+                   RTLD_DI_LINKMAP,
+                   addressof(glx_nvidia_info_ptr)) != 0:
+        return None
+
+    # Grab the contents our of our pointer
+    glx_nvidia_info = cast(glx_nvidia_info_ptr, POINTER(link_map)).contents
+
+    # Decode the path to our library to a str()
+    if glx_nvidia_info.l_name is None:
+        return None
+    try:
+        libglx_nvidia_path = glx_nvidia_info.l_name.decode("utf-8")
```

Comment:
Should this be

```suggestion
        libglx_nvidia_path = os.fsdecode(glx_nvidia_info.l_name)
```

to cope with non-UTF-8 locales?

(This is unlikely to matter in practice, because `libGLX_nvidia.so.0` is almost certainly in an all-ASCII path like `/run/host/usr/lib/x86_64-linux-gnu` or `/run/host/usr/lib64`.)